File: src/main/java/com/cloudpi/cloudpi_backend/user/dto/UserDto.java
package com.cloudpi.cloudpi_backend.user.dto;

import com.cloudpi.cloudpi_backend.security.AccountType;
import com.cloudpi.cloudpi_backend.user.entities.UserEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
public class UserDto {
    private Long id;
    private String username;
    private String email;
    private String nickname;
    private String password;
    private Boolean locked;
    private AccountType userRole;
    private List<String> permissions;
}

File: src/main/java/com/cloudpi/cloudpi_backend/user/dto/UserDto.kt
package com.cloudpi.cloudpi_backend.user.dto;

import com.cloudpi.cloudpi_backend.security.AccountType;
import com.cloudpi.cloudpi_backend.user.entities.UserEntity;

import java.util.List;

public class UserDto {
    private Long id;
    private String username;
    private String email;
    private String nickname;
    private String password;
    private Boolean locked;
    private AccountType userRole;
    private List<String> permissions;
}